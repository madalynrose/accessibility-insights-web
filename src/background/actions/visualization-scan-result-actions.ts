// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Action } from 'common/flux/action';
import { ScanCompletedPayload } from '../../injected/analyzers/analyzer';
import {
    AddTabbedElementPayload,
    AddTabStopInstancePayload,
    RemoveTabStopInstancePayload,
    UpdateTabStopInstancePayload,
    UpdateTabStopRequirementStatusPayload,
} from './action-payloads';

export class VisualizationScanResultActions {
    public readonly scanCompleted = new Action<ScanCompletedPayload<any>>();
    public readonly getCurrentState = new Action();
    public readonly disableIssues = new Action();
    public readonly addTabbedElement = new Action<AddTabbedElementPayload>();
    public readonly disableTabStop = new Action();
    public readonly updateTabStopsRequirementStatus = new Action<
        UpdateTabStopRequirementStatusPayload
    >();
    public readonly addTabStopInstance = new Action<AddTabStopInstancePayload>();
    public readonly updateTabStopInstance = new Action<UpdateTabStopInstancePayload>();
    public readonly removeTabStopInstance = new Action<RemoveTabStopInstancePayload>();
}
