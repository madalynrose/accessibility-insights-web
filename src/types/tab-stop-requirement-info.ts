// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export const TabStopRequirementIds = [
    'keyboard-navigation',
    'keyboard-traps',
    'focus-indicator',
    'tab-order',
    'input-focus',
] as const;

export type TabStopRequirementId = typeof TabStopRequirementIds[number];

export type TabStopRequirementInfo = {
    [requirementId in TabStopRequirementId]: {
        name: string;
        description: string;
    };
};
