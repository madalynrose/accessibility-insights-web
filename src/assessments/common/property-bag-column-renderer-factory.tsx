// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { propertyBagColumnRenderer } from './property-bag-column-renderer';
import * as React from 'react';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: InstanceTableRow<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}

export class PropertyBagColumnRendererWithComputationFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: InstanceTableRow<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRendererWithComputation(item, configs);
        };
    }
}

export function propertyBagColumnRendererWithComputation<TPropertyBag extends ColumnValueBag>(
    item: InstanceTableRow<TPropertyBag>,
    configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
): JSX.Element {
    const mapper = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        if (item.instance.propertyBag == null) {
            return null;
        }
        if (config.compute == null) {
            return null;
        }
        if (
            config.neededPropertyBagValues.some(
                key => item.instance.propertyBag.hasOwnProperty(key) === false,
            ) &&
            config.defaultValue == null
        ) {
            return null;
        }
        return render(config, index);
    };

    const render = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        console.log(item.instance);
        const ValueComponent = config.compute(item.instance.propertyBag);
        console.log('get by calling compute', ValueComponent);
        return (
            <div key={`property-${index}`} className="property-bag-div">
                <span className="display-name">{`${config.displayName}: `}</span>
                <ValueComponent />
            </div>
        );
    };

    return <div className="property-bag-container">{configs.map(mapper)}</div>;
}
