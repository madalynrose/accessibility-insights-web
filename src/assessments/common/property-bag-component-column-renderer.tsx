// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { NamedFC } from 'common/react/named-fc';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { isEmpty } from 'lodash';
import * as React from 'react';

import { DictionaryStringTo } from 'types/common-types';
import { Term } from 'assessments/markup';
import { PropertyBagColumnRenderer } from 'assessments/common/property-bag-column-renderer';

export class PropertyBagComponentColumnRenderer<
    TPropertyBag extends ColumnValueBag,
> extends PropertyBagColumnRenderer<TPropertyBag> {
    private render = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        value: any,
        index: number,
    ) => {
        return (
            <div key={`property-${index}`} className="property-bag-div">
                <span className="display-name">{`${config.displayName}: `}</span>
                {this.renderValue(config, value)}
            </div>
        );
    };

    private mapper = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        if (this.item.instance.propertyBag == null) {
            return null;
        }
        const value = this.item.instance.propertyBag[config.propertyName];
        if (value == null && config.defaultValue == null) {
            return null;
        }

        return this.render(config, value, index);
    };

    protected renderValue = (config: PropertyBagColumnRendererConfig<TPropertyBag>, value: any) => {
        if (config.expand) {
            return this.renderProperties(config, value);
        }
        return <React.Fragment>{() => value}</React.Fragment>;
        // const valueToRender = value || config.defaultValue;
        // if (typeof valueToRender === 'string') {
        //     return <React.Fragment>{valueToRender}</React.Fragment>;
        // }

        // const ValueComponent = valueToRender;
        // console.log(ValueComponent, <ValueComponent />);
        // return (
        //     <React.Fragment>
        //         Hello, this is a <Emphasis>test</Emphasis>
        //     </React.Fragment>
        // );
    };

    private renderProperties = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        propertyMap: DictionaryStringTo<string>,
    ) => {
        if (isEmpty(propertyMap)) {
            return <React.Fragment>{config.defaultValue}</React.Fragment>;
        }

        return Object.keys(propertyMap).map(key => {
            return (
                <div key={key} className="expanded-property-div">
                    {this.renderInnerKeyValue(key, propertyMap[key])}
                </div>
            );
        });
    };

    private renderInnerKeyValue = (key: string, value: string) => {
        if (value === null) {
            return <span className="display-name">{`${key}`}</span>;
        } else {
            return (
                <React.Fragment>
                    <span className="display-name">{`${key}: `}</span>
                    {value}
                </React.Fragment>
            );
        }
    };

    public getRenderer = (): JSX.Element => {
        return <div className="property-bag-container">{this.configs.map(this.mapper)}</div>;
    };
}
