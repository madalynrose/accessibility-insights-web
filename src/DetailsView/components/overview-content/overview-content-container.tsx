// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { NamedFC } from 'common/react/named-fc';
import { HyperlinkDefinition } from 'common/types/hyperlink-definition';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { DetailsViewPivotType } from 'common/types/store-data/details-view-pivot-type';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { TabStoreData } from 'common/types/store-data/tab-store-data';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import * as React from 'react';

import { OverviewSummaryReportModel } from 'reports/assessment-report-model';
import { AssessmentReportSummary } from 'reports/components/assessment-report-summary';
import { GetAssessmentSummaryModelFromProviderAndStoreData } from 'reports/get-assessment-summary-model';
import { GetQuickAssessSummaryModelFromProviderAndStoreData } from 'reports/get-quick-assess-summary-model';

import { TargetChangeDialog, TargetChangeDialogDeps } from '../target-change-dialog';
import styles from './overview-content-container.scss';
import { OverviewHeading } from './overview-heading';
import { OverviewHelpSection, OverviewHelpSectionDeps } from './overview-help-section';

const linkDataSource: HyperlinkDefinition[] = [
    {
        href: 'https://go.microsoft.com/fwlink/?linkid=2082219',
        text: 'Getting started',
    },
    {
        href: 'https://go.microsoft.com/fwlink/?linkid=2082220',
        text: 'How to complete a test',
    },
    {
        href: 'https://go.microsoft.com/fwlink/?linkid=2077941',
        text: 'Ask a question',
    },
    {
        href: 'https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/',
        text: 'New WCAG 2.1 success criteria',
    },
];

export type OverviewContainerDeps = {
    assessmentsProvider: AssessmentsProvider;
    getAssessmentSummaryModelFromProviderAndStoreData: GetAssessmentSummaryModelFromProviderAndStoreData;
    getQuickAssessSummaryModelFromProviderAndStoreData: GetQuickAssessSummaryModelFromProviderAndStoreData;
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
    assessmentsProviderWithFeaturesEnabled: (
        assessmentProvider: AssessmentsProvider,
        flags: FeatureFlagStoreData,
    ) => AssessmentsProvider;
    assessmentsProviderForRequirements: (
        assessmentsProvider: AssessmentsProvider,
        requirementKeys: string[],
    ) => AssessmentsProvider;
    quickAssessRequirementKeys: string[];
} & OverviewHelpSectionDeps &
    TargetChangeDialogDeps;

export interface OverviewContainerProps {
    deps: OverviewContainerDeps;
    assessmentStoreData: AssessmentStoreData;
    tabStoreData: TabStoreData;
    featureFlagStoreData: FeatureFlagStoreData;
    selectedPivot: DetailsViewPivotType;
}

export const overviewContainerAutomationId = 'overviewContainerAutomationId';

export const OverviewContainer = NamedFC<OverviewContainerProps>('OverviewContainer', props => {
    const { deps, assessmentStoreData, tabStoreData, featureFlagStoreData, selectedPivot } = props;
    const {
        assessmentsProvider,
        getAssessmentSummaryModelFromProviderAndStoreData,
        getQuickAssessSummaryModelFromProviderAndStoreData,
        assessmentsProviderWithFeaturesEnabled,
        assessmentsProviderForRequirements,
        quickAssessRequirementKeys,
    } = deps;
    const prevTarget = assessmentStoreData.persistedTabInfo;
    const currentTarget = {
        id: tabStoreData.id,
        url: tabStoreData.url,
        title: tabStoreData.title,
    };
    let filteredProvider = assessmentsProviderWithFeaturesEnabled(
        assessmentsProvider,
        featureFlagStoreData,
    );
    let summaryData: OverviewSummaryReportModel;

    if (selectedPivot === DetailsViewPivotType.assessment) {
        summaryData = getAssessmentSummaryModelFromProviderAndStoreData(
            filteredProvider,
            assessmentStoreData,
        );
    } else {
        filteredProvider = assessmentsProviderForRequirements(
            filteredProvider,
            quickAssessRequirementKeys,
        );
        summaryData = getQuickAssessSummaryModelFromProviderAndStoreData(
            filteredProvider,
            assessmentStoreData,
            quickAssessRequirementKeys,
        );
    }

    return (
        <div data-automation-id={overviewContainerAutomationId} className={styles.overview}>
            <TargetChangeDialog deps={deps} prevTab={prevTarget} newTab={currentTarget} />
            <section className={styles.overviewTextSummarySection}>
                <OverviewHeading />
                <AssessmentReportSummary summary={summaryData} selectedPivot={selectedPivot} />
            </section>
            <section className={styles.overviewHelpSection}>
                <OverviewHelpSection linkDataSource={linkDataSource} deps={deps} />
            </section>
        </div>
    );
});
