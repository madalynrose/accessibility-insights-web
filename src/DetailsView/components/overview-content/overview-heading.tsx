// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { OverviewHeadingIntroFactory } from 'DetailsView/components/overview-content/overview-heading-intro';
import * as React from 'react';

import { NamedFC } from '../../../common/react/named-fc';
import styles from './overview-heading.scss';

export const overviewHeadingAutomationId = 'overview-heading';

export type OverviewHeadingProps = {
    getIntroComponent: OverviewHeadingIntroFactory;
};

export const OverviewHeading = NamedFC('OverviewHeading', (props: OverviewHeadingProps) => {
    return (
        <>
            <div
                className={styles.overviewHeading}
                data-automation-id={overviewHeadingAutomationId}
            >
                <h1>Overview</h1>
                {props.getIntroComponent()}
            </div>
        </>
    );
});
