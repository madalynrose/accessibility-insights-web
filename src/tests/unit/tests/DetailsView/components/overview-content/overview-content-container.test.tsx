// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { shallow } from 'enzyme';
import * as React from 'react';
import { Mock, MockBehavior } from 'typemoq';

import {
    AssessmentStoreData,
    PersistedTabInfo,
} from '../../../../../../common/types/store-data/assessment-result-data';
import { TabStoreData } from '../../../../../../common/types/store-data/tab-store-data';
import { UrlParser } from '../../../../../../common/url-parser';
import { DetailsViewActionMessageCreator } from '../../../../../../DetailsView/actions/details-view-action-message-creator';
import {
    OverviewContainer,
    OverviewContainerDeps,
} from '../../../../../../DetailsView/components/overview-content/overview-content-container';
import { OverviewHelpSectionDeps } from '../../../../../../DetailsView/components/overview-content/overview-help-section';

describe('OverviewContainer', () => {
    const urlParserMock = {} as UrlParser;

    const openExternalLink = jest.fn();

    const tabStoreDataStub: TabStoreData = {
        url: 'some url',
        title: 'some title',
        id: -1,
    } as TabStoreData;

    const overviewHelpSectionDeps = {
        actionInitiators: {
            openExternalLink,
        },
    } as OverviewHelpSectionDeps;

    const assessmentsProvider: AssessmentsProvider = {
        all: () => [],
    } as any;

    const filteredProvider = {} as AssessmentsProvider;
    const detailsViewActionMessageCreator = {} as DetailsViewActionMessageCreator;
    const assessmentsProviderWithFeaturesEnabledMock = Mock.ofInstance(
        (provider, featureFlagData) => null,
        MockBehavior.Strict,
    );
    const assessmentsProviderForRequirementsMock = Mock.ofInstance(
        (provider, requirements) => null,
        MockBehavior.Strict,
    );
    const getFilteredProviderMock = Mock.ofInstance(props => null, MockBehavior.Strict);
    const getSummaryModelFromProviderAndStoreDataMock = Mock.ofInstance(
        props => null,
        MockBehavior.Strict,
    );
    const getAssessmentSummaryModelFromProviderAndStoreData = jest.fn();
    const getQuickAssessSummaryModelFromProviderAndStoreData = jest.fn();
    const quickAssessRequirementKeysStub = [];
    const deps: OverviewContainerDeps = {
        assessmentsProvider: assessmentsProvider,
        actionInitiators: overviewHelpSectionDeps.actionInitiators,
        getAssessmentSummaryModelFromProviderAndStoreData:
            getAssessmentSummaryModelFromProviderAndStoreData,
        getQuickAssessSummaryModelFromProviderAndStoreData:
            getQuickAssessSummaryModelFromProviderAndStoreData,
        detailsViewActionMessageCreator,
        urlParser: urlParserMock,
        assessmentsProviderWithFeaturesEnabled: assessmentsProviderWithFeaturesEnabledMock.object,
        assessmentsProviderForRequirements: assessmentsProviderForRequirementsMock.object,
        detailsViewId: undefined,
        quickAssessRequirementKeys: quickAssessRequirementKeysStub,
    };

    const featureFlagDataStub = {};

    const assessmentStoreData: AssessmentStoreData = {
        persistedTabInfo: {} as PersistedTabInfo,
    } as AssessmentStoreData;

    getFilteredProviderMock
        .setup(mock => mock({ deps, featureFlagStoreData: featureFlagDataStub }))
        .returns(() => filteredProvider);
    getSummaryModelFromProviderAndStoreDataMock.setup(mock =>
        mock({ deps, assessmentsProvider: filteredProvider, assessmentStoreData }),
    );

    let component: JSX.Element;

    it('component is defined and matches snapshot', () => {
        component = (
            <OverviewContainer
                deps={deps}
                assessmentStoreData={assessmentStoreData}
                featureFlagStoreData={featureFlagDataStub}
                tabStoreData={tabStoreDataStub}
                getFilteredProvider={getFilteredProviderMock.object}
                getSummaryModelFromProviderAndStoreData={
                    getSummaryModelFromProviderAndStoreDataMock.object
                }
            />
        );
        const wrapper = shallow(component);
        expect(component).toBeDefined();
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
