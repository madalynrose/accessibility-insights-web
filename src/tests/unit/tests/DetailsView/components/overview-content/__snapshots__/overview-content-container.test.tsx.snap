// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewContainer component is defined and matches snapshot 1`] = `
<div
  className="overview"
  data-automation-id="overviewContainerAutomationId"
>
  <TargetChangeDialog
    deps={
      {
        "actionInitiators": {
          "openExternalLink": [MockFunction],
        },
        "assessmentsProvider": {
          "all": [Function],
        },
        "assessmentsProviderForRequirements": [Function],
        "assessmentsProviderWithFeaturesEnabled": [Function],
        "detailsViewActionMessageCreator": {},
        "detailsViewId": undefined,
        "getAssessmentSummaryModelFromProviderAndStoreData": [MockFunction],
        "getQuickAssessSummaryModelFromProviderAndStoreData": [MockFunction],
        "quickAssessRequirementKeys": [],
        "urlParser": {},
      }
    }
    newTab={
      {
        "id": -1,
        "title": "some title",
        "url": "some url",
      }
    }
    prevTab={{}}
  />
  <section
    className="overviewTextSummarySection"
  >
    <OverviewHeading />
    <AssessmentReportSummary />
  </section>
  <section
    className="overviewHelpSection"
  >
    <OverviewHelpSection
      deps={
        {
          "actionInitiators": {
            "openExternalLink": [MockFunction],
          },
          "assessmentsProvider": {
            "all": [Function],
          },
          "assessmentsProviderForRequirements": [Function],
          "assessmentsProviderWithFeaturesEnabled": [Function],
          "detailsViewActionMessageCreator": {},
          "detailsViewId": undefined,
          "getAssessmentSummaryModelFromProviderAndStoreData": [MockFunction],
          "getQuickAssessSummaryModelFromProviderAndStoreData": [MockFunction],
          "quickAssessRequirementKeys": [],
          "urlParser": {},
        }
      }
      linkDataSource={
        [
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2082219",
            "text": "Getting started",
          },
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2082220",
            "text": "How to complete a test",
          },
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2077941",
            "text": "Ask a question",
          },
          {
            "href": "https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/",
            "text": "New WCAG 2.1 success criteria",
          },
        ]
      }
    />
  </section>
</div>
`;

exports[`OverviewContainer component is defined and matches snapshot for pivotType 2 1`] = `
<div
  className="overview"
  data-automation-id="overviewContainerAutomationId"
>
  <TargetChangeDialog
    deps={
      {
        "actionInitiators": {
          "openExternalLink": [MockFunction],
        },
        "assessmentsProvider": {
          "all": [Function],
        },
        "assessmentsProviderForRequirements": [Function],
        "assessmentsProviderWithFeaturesEnabled": [Function],
        "detailsViewActionMessageCreator": {},
        "detailsViewId": undefined,
        "getAssessmentSummaryModelFromProviderAndStoreData": [MockFunction] {
          "calls": [
            [
              {},
              {
                "persistedTabInfo": {},
              },
            ],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "getQuickAssessSummaryModelFromProviderAndStoreData": [MockFunction],
        "quickAssessRequirementKeys": [],
        "urlParser": {},
      }
    }
    newTab={
      {
        "id": -1,
        "title": "some title",
        "url": "some url",
      }
    }
    prevTab={{}}
  />
  <section
    className="overviewTextSummarySection"
  >
    <OverviewHeading />
    <AssessmentReportSummary
      selectedPivot={2}
    />
  </section>
  <section
    className="overviewHelpSection"
  >
    <OverviewHelpSection
      deps={
        {
          "actionInitiators": {
            "openExternalLink": [MockFunction],
          },
          "assessmentsProvider": {
            "all": [Function],
          },
          "assessmentsProviderForRequirements": [Function],
          "assessmentsProviderWithFeaturesEnabled": [Function],
          "detailsViewActionMessageCreator": {},
          "detailsViewId": undefined,
          "getAssessmentSummaryModelFromProviderAndStoreData": [MockFunction] {
            "calls": [
              [
                {},
                {
                  "persistedTabInfo": {},
                },
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "getQuickAssessSummaryModelFromProviderAndStoreData": [MockFunction],
          "quickAssessRequirementKeys": [],
          "urlParser": {},
        }
      }
      linkDataSource={
        [
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2082219",
            "text": "Getting started",
          },
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2082220",
            "text": "How to complete a test",
          },
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2077941",
            "text": "Ask a question",
          },
          {
            "href": "https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/",
            "text": "New WCAG 2.1 success criteria",
          },
        ]
      }
    />
  </section>
</div>
`;

exports[`OverviewContainer component is defined and matches snapshot for pivotType 3 1`] = `
<div
  className="overview"
  data-automation-id="overviewContainerAutomationId"
>
  <TargetChangeDialog
    deps={
      {
        "actionInitiators": {
          "openExternalLink": [MockFunction],
        },
        "assessmentsProvider": {
          "all": [Function],
        },
        "assessmentsProviderForRequirements": [Function],
        "assessmentsProviderWithFeaturesEnabled": [Function],
        "detailsViewActionMessageCreator": {},
        "detailsViewId": undefined,
        "getAssessmentSummaryModelFromProviderAndStoreData": [MockFunction],
        "getQuickAssessSummaryModelFromProviderAndStoreData": [MockFunction] {
          "calls": [
            [
              {},
              {
                "persistedTabInfo": {},
              },
              [],
            ],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "quickAssessRequirementKeys": [],
        "urlParser": {},
      }
    }
    newTab={
      {
        "id": -1,
        "title": "some title",
        "url": "some url",
      }
    }
    prevTab={{}}
  />
  <section
    className="overviewTextSummarySection"
  >
    <OverviewHeading />
    <AssessmentReportSummary
      selectedPivot={3}
    />
  </section>
  <section
    className="overviewHelpSection"
  >
    <OverviewHelpSection
      deps={
        {
          "actionInitiators": {
            "openExternalLink": [MockFunction],
          },
          "assessmentsProvider": {
            "all": [Function],
          },
          "assessmentsProviderForRequirements": [Function],
          "assessmentsProviderWithFeaturesEnabled": [Function],
          "detailsViewActionMessageCreator": {},
          "detailsViewId": undefined,
          "getAssessmentSummaryModelFromProviderAndStoreData": [MockFunction],
          "getQuickAssessSummaryModelFromProviderAndStoreData": [MockFunction] {
            "calls": [
              [
                {},
                {
                  "persistedTabInfo": {},
                },
                [],
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "quickAssessRequirementKeys": [],
          "urlParser": {},
        }
      }
      linkDataSource={
        [
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2082219",
            "text": "Getting started",
          },
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2082220",
            "text": "How to complete a test",
          },
          {
            "href": "https://go.microsoft.com/fwlink/?linkid=2077941",
            "text": "Ask a question",
          },
          {
            "href": "https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/",
            "text": "New WCAG 2.1 success criteria",
          },
        ]
      }
    />
  </section>
</div>
`;
